pkg/service/credential/credential.go

```diff
@@ -168,6 +175,51 @@ func (s Service) signCredentialJWT(issuer string, cred credential.VerifiableCred
 	return &credToken.Token, nil
 }
 
+type VerifyCredentialRequest struct {
+	DataIntegrityCredential *credential.VerifiableCredential `json:"credential,omitempty"`
+	CredentialJWT           *string                          `json:"credentialJwt,omitempty"`
+}
+
+func (vcr VerifyCredentialRequest) IsValid() error {
+	if vcr.DataIntegrityCredential == nil && vcr.CredentialJWT == nil {
+		return errors.New("either a credential or a credential JWT must be provided")
+	}
+	if vcr.DataIntegrityCredential != nil && vcr.CredentialJWT != nil {
+		return errors.New("only one of credential or credential JWT can be provided")
+	}
+	return nil
+}
+
+type VerifyCredentialResponse struct {
+	Verified bool   `json:"verified" json:"verified"`
+	Reason   string `json:"reason,omitempty" json:"reason,omitempty"`
+}
+
+// VerifyCredential does three levels of verification on a credential:
+// 1. Makes sure the credential has a valid signature
+// 2. Makes sure the credential has is not expired
+// 3. Makes sure the credential complies with the VC Data Model
+// 4. If the credential has a schema, makes sure its data complies with the schema
+// LATER: Makes sure the credential has not been revoked, other checks.
+// Note: https://github.com/TBD54566975/ssi-sdk/issues/213
+func (s Service) VerifyCredential(request VerifyCredentialRequest) (*VerifyCredentialResponse, error) {
+	if err := request.IsValid(); err != nil {
+		return nil, util.LoggingErrorMsg(err, "invalid verify credential request")
+	}
+
+	if request.CredentialJWT != nil {
+		if err := s.verifier.VerifyJWTCredential(*request.CredentialJWT); err != nil {
```

Comment:
Could be cool for audit logging or dev sanity to have a function (or debug flag) that would log how many validatioons were run.

Would be terrible if only expiration date validator was applied to this and then we think its 'fully verified' but only a portion.

Maybe there should be a 'FullyValidate' function that runs all validators ?

Reply:
let me add the logging and think about fully validate

Reply:
the verify function is 'fully validate' maybe the name is confusing

as per your other comment I opened an issue here https://github.com/TBD54566975/ssi-service/issues/122